src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    DeviceInfo,
+    PackageInfo,
+    PermissionInfo,
+} from 'electron/platform/android/android-service-configurator';
+
+type AdbDevice = {
+    udid: string;
+};
+
+const servicePackageName: string = 'com.microsoft.accessibilityinsightsforandroidservice';
+
+export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
+    private readonly adb: ADB;
+
+    constructor(adb: ADB) {
+        this.adb = adb;
+    }
+
+    public getConnectedDevices = async (): Promise<Array<DeviceInfo>> => {
+        // First add devices that are flagged as emulator, then
+        // add devices that aren't flagged as emulators
+        const detectedDevices = {};
+        await this.addDevices(await this.adb.getConnectedEmulators(), true, detectedDevices);
+        await this.addDevices(await this.adb.getConnectedDevices(), false, detectedDevices);
+
+        const output: Array<DeviceInfo> = [];
+        Object.values(detectedDevices).forEach(device => output.push(device as DeviceInfo));
+        return output;
+    };
+
+    private async addDevices(
+        devices: Array<AdbDevice>,
+        isEmulator: boolean,
+        detectedDevices: any,
```

Comment:
This can be given a type (either `{ [udid: string]: DeviceInfo }` or `DictionaryStringTo<DeviceInfo>`, whichever you find more readable)

Reply:
`DictionaryStringTo` is closest to what I'm used to